### ISSUE TITLE
Separate Gulpfiles for different project sections, unify with exec

### ISSUE THREAD

**Issue by sreynen** [AtenDesignGroup/aten-gulp#1]
I'm typically editing a very specific section of a project, e.g. a single (output) JS file. When using a single Gulpfile for the entire project, my options are either:

1) Run the full `dev` command.
2) Hunt through the Gulpfile for the command specific to what I'm doing.

I suggest we make this easier by putting each section in a separate Gulpfile. Then editing a specific file only requires running the nearest Gulpfile.

By itself, that would make project-wide changes a pain, as those would require running multiple Gulpfiles. Fortunately Gulp can run Gulpfiles, so we could maintain the current functionality with a master Gulpfile that does nothing but run the other Gulpfiles.

**Comment by Josh-Miller** [comment]
I think another additional way we could fix this is by documenting our gulp functions, maybe creating a flag like: gulp --help

**Comment by jnettik** [comment]
Would this be something that's also solvable by standardization and documentation?

**Comment by jnettik** [comment]
Gah! Josh beat me...

**Comment by Josh-Miller** [comment]
Looks like a --help flag will be in v4: https://github.com/gulpjs/gulp/issues/1104

...or we could create a plugin for it.

**Comment by sreynen** [comment]
Yeah, standardizing and/or documenting the commands would resolve the problem of figuring out how to target them. I'm still partial to having multiple files. Some additional advantages:

1) Editing Gulpfiles is also more focused.
2) The code Gulpfiles are touching is more self-contained.

What are the disadvantages? Compile speed?

**Comment by Josh-Miller** [comment]
I doubt it affects compile speed, but we'd have to manage more file path configuration. Plus we'd have to dig through folders to make any changes to our gulpfiles, whereas, now there is a central place. But other than personal structure choices I don't think there would be any affects for the compiling itself.

**Comment by jnettik** [comment]
Scott was showing me how he has his gulp stuff organized. What I liked about it is it mirrors a distinction I've been trying to draw between "commands" and "tasks". 

A command is something like `gulp styles`, but depending on the needs of the project that could fire off tasks for Sass, Less, Stylus, Linting, Prefixing, etc. Plus that way it makes the code around what commands are available more self documenting as it's all in the main `Gulpfile.js`.

**Comment by sreynen** [comment]
Here's the master Gulpfile James mentioned, which I'm using for cinc.io:

```
var gulp = require('gulp'),
    notify = require('gulp-notify')
    exec = require('child_process').exec;

gulp.task('default', function() {
  gulp.start('css', 'js', 'news');
});

gulp.task('css', function() {
  exec('cd ../css-make && gulp && cd ../make', function (err, stdout, stderr) {
    console.log(stdout);
    console.log(stderr);
  });
});

gulp.task('js', function() {
  exec('cd ../js-make && gulp && cd ../make', function (err, stdout, stderr) {
    console.log(stdout);
    console.log(stderr);
  });
});

gulp.task('news', function() {
  exec('cd ../news-make && metalsmith && cd ../make', function (err, stdout, stderr) {
    console.log(stdout);
    console.log(stderr);
  });
});
```

**Comment by pixelwhip** [comment]
I have the same concerns as Josh about managing paths.  Not that it's handled perfectly now.  I do like that I can edit them all in one place. Taking a D7 site as an example. Where would the various Gulpfiles live with this method?  `sites/all/modules`? `sites/all/modules/name-of-module`?

My other concern is with a task like `gulp styles` firing off a bunch of sub-tasks like sass, linting and auto prefixing. What's the advantage of running Sass without auto-prefixer. Wouldn't that mean we have to save the files incrementally, losing Gulp's speed advantage of streaming files through processes?
